nixos/modules/services/misc/workout-tracker.nix

```diff
@@ -0,0 +1,121 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+with lib;
+
+let
+  inherit (lib) types;
+  cfg = config.services.workout-tracker;
+  exe = lib.getExe cfg.package;
+in
+
+{
+  options = {
+    services.workout-tracker = {
+      enable = lib.mkEnableOption "A workout tracking web application for personal use (or family, friends), geared towards running and other GPX-based activities";
+
+      package = lib.mkPackageOption pkgs "workout-tracker" { };
+
+      address = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = lib.mdDoc "Web interface address.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 8080;
+        description = lib.mdDoc "Web interface port.";
+      };
```

Comment:
I think it should be possible to include these in a RFC42 compliant settings option (you can specify the attribute explicitly there with a default)

Reply:
I think I made the changes as laid out in `settings-optiona.sections.md` now, but please check. I have not written a module before.

Reply:
This is not really what I meant unfortunately. I meant that you define a `settings` option (as you did) and then autogenerate the full environment for the systemd service from it. You can find one example for this here: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/paperless.nix#L30-L34

Reply:
Okay, I adopted solution from a different module, but please check.